### PR TITLE
Fix a issue in REPL related to user-defined sorts and lemmas

### PR DESCRIPTION
**Pull request by yzhang90**
…red in SMT

---

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

**Comment by yzhang90** [comment]
@ttuegel @virgil-serbanuta I made a fix to REPL so that user-defined sorts and lemmas are declared in the SMT queries. However, i'm not sure how to write a test.

File: kore/app/repl/Main.hs [hunk 1]
```diff
@@ -212,13 +217,15 @@ mainWithOptions
...
                 $ SMT.runSMT smtConfig (swappableLogger mLogger)
-                   $ proveWithRepl
-                        indexedModule
-                        specDefIndexedModule
-                        mLogger
-                        replScript
-                        replMode
-                        outputFile
+                   $ do
+                        give (MetadataTools.build indexedModule) (declareSMTLemmas indexedModule)
```
Comment (virgil-serbanuta) [inline kore/app/repl/Main.hs]: Lines should usually be < 80 chars.

**Comment by ana-pantilie** [comment]
@yzhang90 
Yes, the `scripts` directory from the link @virgil-serbanuta added contains `repl` integration tests. You can find more about `repl` scripts here https://github.com/kframework/kore/wiki/Kore-REPL#repl-scripts

We check the `repl` output after running each script against its corresponding `.golden` output file. The `repl` outputs a new conjuction of claims which indicate the state of the proof after executing the script. If the proof was finished the output is `#True`. You can look at the `tiny` semantics directory for how to add this sort of integration test. Each spec should have its own directory and a `scripts` directory which tests proofs of that spec.

So, if you think that you can test this by checking the final repl output (the set of claims) after running some repl commands on a spec and semantics of your choice you can add this sort of test. Otherwise, we also have some unit tests for the parser and the interpreter in https://github.com/kframework/kore/tree/master/kore/test/Test/Kore/Repl, if you think this sort of test would be more suitable in this case.

I hope this helps.

File: kore/app/repl/Main.hs (hunk 1 above)
Comment (yzhang90) [reply kore/app/repl/Main.hs]: fixed

**Comment by yzhang90** [comment]
@virgil-serbanuta I added a TODO in the code. I'm busy with the sum-to-n proof and we can get back to the test later when I have time.

**Pull request merged**